Assets/Scripts/PlayerController.cs

```diff
@@ -21,28 +41,31 @@ public class PlayerController : MonoBehaviour {
     [SerializeField]
     private ContactFilter2D groundContactFilter;
     [SerializeField]
-    private Text collectableText;
-    [SerializeField]
     private Slider overheatSlider;
 
-
     private Rigidbody2D rb2d;
     private Animator anim;
+    private AudioSource[] audioSources;
```

Comment:
The player more than likely only represents a single audio source. Audio sources should be considered like speakers in the scene. You wouldn't need a new set of speakers for every song you want to play at home. You only need new speakers in the positions in the world that are important for positional audio. Since all these sounds in theory are coming from the player, a single audio source component makes sense.

Instead of multiple sources, us multiple clips and a single source. See my next comment below.